From 3607c542dc42fe11beda9866ce178616d62625c7 Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Sun, 02 May 2021 09:37:00 +0300
Subject: [PATCH] [SW-34387] gaudi2: disable arc initialization

Boot FIT image contains zephyr mgmt and pid images
besides the ARM uboot. Until the actual zephyr
support is made available, dummy "all zero"
binaries are created, as place holders. So,
skip booting the ARCs until valid zephyr images
are made available.

Change-Id: I23b26857fb7152d4f4c98c0843fda59b37b6766b
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/gaudi2/common.c b/gaudi2/common.c
index 134503e..704a533 100644
--- a/gaudi2/common.c
+++ b/gaudi2/common.c
@@ -405,6 +405,9 @@
 
 static void init_arc(void)
 {
+	/* TODO: Halt and reset ARCs once zephyr support is added */
+	return;
+
 	HALT_N_RESET_ARC(0);
 	HALT_N_RESET_ARC(1);
 
@@ -495,19 +498,27 @@
 
 void preboot_start_image(u64 startaddr)
 {
-	/* Set ARC0 and ARC1 in running state */
-	RUN_ARC(0);
-	RUN_ARC(1);
-
 	/*
 	 * After image_entry, control is not expected to return
 	 * here and continue.
 	 */
 	image_start(startaddr);
+
+
+	/* Set ARC0 and ARC1 in running state */
+	/*
+	 * TODO: Move run arc above image start once zephyr
+	 * support is added
+	 */
+	RUN_ARC(0);
+	RUN_ARC(1);
 }
 
 static void fit_copy_arc_image(u64 dst, const void *buf, int len)
 {
+	/* TODO: copy images once zephyr support is added */
+	return;
+
 	/*
 	 * initialize scratch SRAM to avoid ECC exceptions
 	 * ECC errors are observed due to byte copy to SP
